.github/workflows/nightly.yml

```diff
@@ -1,121 +1,156 @@
-    
-    - name: Add -nightly+(date)-(commit ID) prefix to version
-      id: set-nightly-version
+    - name: Choose base commit for nightly snapshot
+      id: choose-base
+      env:
+        GH_EVENT_NAME: ${{ github.event_name }}
       run: |
-        echo version=${{ steps.set-version.outputs.version }}-nightly+`date +%F`-`git rev-parse --short HEAD` >> "$GITHUB_OUTPUT"
+        echo "Event triggered by $GH_EVENT_NAME"
+        echo
 
-    - name: Log new version & changelog
-      run: |
-        echo "Proposed new version: $VERSION"
-        echo "Nightly version: $NIGHTLY_VERSION"
-      env:
-        VERSION: ${{ steps.set-version.outputs.version }}
-        NIGHTLY_VERSION: ${{ steps.set-nightly-version.outputs.version }}
+        echo "Listing branches on remote"
+        echo
+
+        git --no-pager branch --remote | sed 's|origin/||'
+
+        echo
+
+        git checkout -b nightly --track origin/nightly
+        OLD_BASE_COMMIT=$(git log nightly --skip=1 -n 1 --format=%H)
+
+        echo "Nightly commit ID is $OLD_BASE_COMMIT"
+
+        LATEST_RELEASE_PLZ=$(git --no-pager branch --format="%(refname:short)" --remote |
+          sed 's|origin/||' |
+          grep -E '^release-plz-20' |
+          sort -r |
+          head -n 1)
+
+        if [ -z "$LATEST_RELEASE_PLZ" ]; then
+
+          echo "No release-plz branch; using main"
+          git checkout -b main --track main
+          BASE_BRANCH=main;
+
+        else
+
+          echo "Found RP branch $LATEST_RELEASE_PLZ; comparing against main"
+
+          git checkout -b "$LATEST_RELEASE_PLZ" --track "origin/$LATEST_RELEASE_PLZ"
+
+          if git show-ref --verify --quiet refs/heads/main; then
+            echo "Branch 'main' already exists locally";
+          else
+            git checkout -b main --track origin/main;
+          fi
+
+          export RP_TS=$(git log -1 --format=%cd --date=iso-strict $LATEST_RELEASE_PLZ)
+          export MAIN_TS=$(git log -1 --format=%cd --date=iso-strict main)
+
+          echo "Branch $LATEST_RELEASE_PLZ updated at $RP_TS"
+          echo "Branch main updated at $MAIN_TS"
+
+          if [[ "$RP_TS" > "$MAIN_TS" ]]; then
+            git checkout $LATEST_RELEASE_PLZ
+            BASE_BRANCH=$LATEST_RELEASE_PLZ;
+          else
+            git checkout main
+            BASE_BRANCH=main;
+          fi
+
+        fi
+
+        NEW_BASE_COMMIT=$(git log $BASE_BRANCH -n 1 --format=%H)
+
+        echo "Choosing base branch $BASE_BRANCH because it's newer"
+        echo "$BASE_BRANCH commit ID is $NEW_BASE_COMMIT"
+
+        if [[ "$OLD_BASE_COMMIT" == "$NEW_BASE_COMMIT" && "$GH_EVENT_NAME" == "pull_request" ]]; then
+          echo "Nightly already points to same base; leave unchanged";
+          echo "base-commit=skip" >> "$GITHUB_OUTPUT"
+          exit 0
+        elif [[ "$GH_EVENT_NAME" == "push" ]]; then
+          LATEST_TAG=$(git describe --tags --abbrev=0)
+
+          echo Looking for non-chore commits since last tag
```

Comment:
This logic is a bit tricky. The goal here is to see if `main` has any `feat` or `fix` commits since the previous release.

If so, then we expect release-plz to generate a new release PR and we'll wait for that update before generating a new nightly branch. (Reason for that is that we want to use the same version number(s) that release-plz is proposing.)

If there are no new commits or they are all `chore` type commits, then release-plz will _not_ generate a release PR and we want to immediately generate a new nightly snapshot based on the latest commit in `main`.